wallet/src/repository/wallet/mod.rs

```diff
@@ -1001,6 +1016,83 @@ where
         self.create_transaction_components(state, value, fee, recipient, false)
     }
 
+    fn create_vt_body(
+        &self,
+        state: &mut State,
+        value: u64,
+        fee: u64,
+        recipient: Option<(types::PublicKeyHash, u64)>,
+    ) -> Result<(types::VTTransactionBody, types::TransactionComponents)> {
+        // Set this to 1-2 output.
+        let max_iterations = self.params.max_vt_weight / types::INPUT_SIZE;
+
+        let mut initial_fee =
+            u64::from(types::INPUT_SIZE + 2 * types::OUTPUT_SIZE * types::GAMMA) * fee;
+        for _ in 0..max_iterations {
```

Comment:
Can we explain in a comment what this algorithm is trying to do (specially in each iteration)?

Reply:
Absolutely